src/osd/PrimaryLogPG.cc

```diff
@@ -739,10 +739,14 @@ void PrimaryLogPG::maybe_force_recovery()
     if (*it == get_primary()) continue;
     pg_shard_t peer = *it;
     if (peer_missing.count(peer) &&
-	!peer_missing[peer].get_items().empty() &&
-	min_version > peer_missing[peer].get_rmissing().begin()->first) {
-      min_version = peer_missing[peer].get_rmissing().begin()->first;
-      soid = peer_missing[peer].get_rmissing().begin()->second;
+	!peer_missing[peer].get_items().empty()) {
```

Comment:
might want to drop this change or split it into another separate "refactor only" commit.